### PR TITLE
docs(cli): improve upgrade message

### PR DESCRIPTION
**Pull request by blimmer**
This PR relates to #18024. It adds information on upgrading between major versions to the upgrade banner.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk/lib/util/npm.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { exec as _exec } from 'child_process';
+import { promisify } from 'util';
+import * as semver from 'semver';
+import { debug } from '../../lib/logging';
+
+const exec = promisify(_exec);
+
+export async function getLatestVersionFromNpm(): Promise<string> {
```
Comment (blimmer) [inline packages/aws-cdk/lib/util/npm.ts:8]: I broke this method out into a separate file/module so I can easily stub it in tests. It's challenging/confusing to mock a method in the same file with Jest or Sinon (see https://stackoverflow.com/questions/45111198/how-to-mock-functions-in-the-same-module-using-jest and https://stackoverflow.com/questions/39861674/stubbing-method-in-same-file-using-sinon)

File: packages/aws-cdk/lib/version.ts [hunk 2]
```diff
@@ -97,19 +91,30 @@ export async function latestVersionIfHigher(currentVersion: string, cacheFile: V
...
+
+export async function displayVersionMessage(currentVersion = versionNumber(), versionCheckCache?: VersionCheckTTL): Promise<void> {
   if (!process.stdout.isTTY || process.env.CDK_DISABLE_VERSION_CHECK) {
     return;
   }
 
   try {
-    const versionCheckCache = new VersionCheckTTL();
-    const laterVersion = await latestVersionIfHigher(versionNumber(), versionCheckCache);
+    const laterVersion = await latestVersionIfHigher(currentVersion, versionCheckCache ?? new VersionCheckTTL());
```
Comment (blimmer) [inline packages/aws-cdk/lib/version.ts:115]: I added the ability to pass a version check file via dependency injection to this method to add additional tests. This ensures that the tests aren't mucking with the global version check file.

File: packages/aws-cdk/test/version.test.ts [hunk 3]
```diff
@@ -77,9 +78,49 @@ test('No Version specified for storage in the TTL file', async () => {
 });
 
 test('Skip version check if environment variable is set', async () => {
-  process.stdout.isTTY = true;
-  process.env.CDK_DISABLE_VERSION_CHECK = '1';
+  sinon.stub(process, 'stdout').value({ ...process.stdout, isTTY: true });
+  sinon.stub(process, 'env').value({ ...process.env, CDK_DISABLE_VERSION_CHECK: '1' });
```
Comment (blimmer) [inline packages/aws-cdk/test/version.test.ts:82]: I changed this test from directly setting these global "process" variables to stubbing them with sinon. This ensures that the values get set back to their original values after the tests run. Before this change, the `CDK_DISABLE_VERSION_CHECK` environment variable was still set in the environment, causing my new tests below to fail.

**Review by rix0rrr — APPROVED** [review]
This is great, thanks!

**Pull request merged**